File: solutions/python/intervals/remove-covered-intervals.py
# LeetCode 1288 — Remove Covered Intervals (Medium)
# Category: Intervals · Approach: Sort + sweep
# Time: O(n log n) | Space: O(1)
# Source: https://leetcode.com/problems/remove-covered-intervals/

def remove_covered_intervals(intervals: list[list[int]]) -> int:
    intervals.sort(key=lambda iv: (iv[0], -iv[1]))
    count = 0
    prev_end = 0
    for _, end in intervals:
        if end > prev_end:
            count += 1
            prev_end = end
    return count

File: solutions/typescript/intervals/remove-covered-intervals.ts
// LeetCode 1288 — Remove Covered Intervals (Medium)
// Category: Intervals · Approach: Sort + sweep
// Time: O(n log n) | Space: O(1)
// Source: https://leetcode.com/problems/remove-covered-intervals/

function removeCoveredIntervals(intervals: number[][]): number {
  intervals.sort((a, b) => (a[0] - b[0]) || (b[1] - a[1]));
  let count = 0;
  let prevEnd = 0;
  for (const [, end] of intervals) {
    if (end > prevEnd) {
      count += 1;
      prevEnd = end;
    }
  }
  return count;
}
